public/apps/accountinfo/accountinfo.js

```diff
@@ -45,26 +47,64 @@ uiRoutes
         controllerAs: 'ctrl'
     });
 
-uiModules
-    .get('app/security-accountinfo')
-    .controller('accountInfoNavController', function ($http, $window, Private, security_resolvedInfo) {
+const app = uiModules.get('app/security-accountinfo', []);
 
-        var APP_ROOT = `${chrome.getBasePath()}`;
-        var API_ROOT = `${APP_ROOT}/api/v1`;
+app.controller('accountInfoNavController', function ($scope, $http, $window, Private, security_resolvedInfo, backendAccount) {
 
-        this.opendistro_security_version = chrome.getInjected("opendistro_security_version");
+    $scope.security_user = {}
+    $scope.service = backendAccount;
+    $scope.resourcelabel = "Account";
+    $scope.show_password_box = false;
 
+    this.is_basicauth = chrome.getInjected("auth.type") == 'basicauth';
+    this.opendistro_security_version = chrome.getInjected("opendistro_security_version");
 
-        $http.get(`${API_ROOT}/auth/authinfo`)
-            .then(
-                (response) => {
+    $scope.service.fetch($scope.resource)
+        .then(
+          (response) => {
+            $scope.service.postFetch(response);
+            $scope.security_user = response.data;
+            $scope.show_password_box = this.is_basicauth && !$scope.security_user.is_reserved;
+            $scope.resourcename = $scope.security_user.user_name;
+          },
+          (error) => {
+            toastNotifications.addDanger({
+                text: error.data.message,
+            });
+          }
+        );
+});
+
+app.controller('accountEditController', function ($scope, $element, $route, $location, $routeParams, createNotifier, backendAccount, kbnUrl, securityAccessControl) {
+    $scope.service = backendAccount;
+
+    $scope.saveObject = (event) => {
+        if (event) {
+            event.preventDefault();
+        }
+
+        const form = $element.find('form[name="objectForm"]');
+
+        if (form.hasClass('ng-invalid-required')) {
+            $scope.errorMessage = 'Please fill in all the required parameters.';
+            return;
+        }
+        if (!form.hasClass('ng-valid')) {
+            $scope.errorMessage = 'Please correct all errors and try again.';
+            return;
+        }
 
-                    this.security_user = response.data;
-                },
-                (error) => {
-                    toastNotifications.addDanger({
-                        text: error.data.message,
-                    });
-                }
+        if ($scope.resource.password !== $scope.resource.password_confirmation) {
+            $scope.errorMessage = 'Passwords do not match.';
+            return;
+        }
+
+        $scope.service.save($scope.resource)
+            .then(
+                () => { securityAccessControl.logout() },
```

Comment:
Should we prompt the user that we will log him out if we change the password successfully? Maybe show a success message and then redirect.

Reply:
Currently it shows a toast telling "Account saved" briefly and then logs out. 
Looks like elastic gives proper support for modals through react only. https://elastic.github.io/eui/#/layout/modal 
Wondering if it ll be easier to add a tooltip or info icon to password box. What are your thoughts of doing this ?

Reply:
Toast message should suffice. Just out of curiousity, can we update cookies/headers with the new password once its updated? It may prevent the need to logout and re-login.

Reply:
Invalidating the cookie would require hacks as re-auth with the backend must be done again  in the middleware and inject the new user for both client and backend. Any intermediate request will hit 403 and auto-logout. I was wondering if this could be removed from the scope of this PR

Reply:
Sure.